lib/index.d.ts

```diff
@@ -0,0 +1,506 @@
+// If `racer` were written as an ES module, then it would have a default export
+// that's an instance of Racer, plus exports of module classes/types like Model.
+//
+// However, this is a CommonJS module, and to correctly model it in TypeScript,
+// we'd have to do `const racer = new Racer(); export = racer;`. Unfortunately,
+// as of TypeScript 3.4, we can't merge namespaces into object variables:
+// https://github.com/Microsoft/TypeScript/issues/18163
+//
+// That means we can't `export =` a Racer instance and also export
+// classes/types. To work around this, we simulate the important Racer instance
+// methods on the exported namespace.
+export = racer;
+
+declare namespace racer {
+  // Simulate important Racer instance methods.
+
+  /** Creates a new RacerBackend. Only available on the server. */
+  export function createBackend(options?: ShareBackendOptions & {modelOptions?: ModelOptions}): RacerBackend;
+
+  export function createModel(data?: ModelBundle): RootModel;
+
+  // https://github.com/share/sharedb/blob/master/lib/backend.js
+  interface ShareBackendOptions {
+    db?: any;
+    extraDbs?: {[extraDbName: string]: any};
+    pubsub?: any;
+
+    disableDocAction?: boolean;
+    disableSpaceDelimitedActions?: boolean;
+    maxSubmitRetries?: number;
+    suppressPublish?: boolean;
+  }
+
+  //
+  //     backend.js
+  //
+
+  class RacerBackend {
```

Comment:
Should this class include `createBackend` and relevant properties?

Using the API like so:
```js
const backend: RacerBackend = racer.createBackend({db: ... })
// ... in another function
{ db } = backend.db
```

Reply:
As discussed in person:
* `createBackend` is on the `racer` module, not on RacerBackend, and the definitions do that.
* The `backend.db` is "inherited" from `sharedb.Backend`, and it's an instance of a `sharedb.Db` (e.g. an instance of sharedb-mongo).

For the second point, I've pushed a commit to have `RacerBackend`'s definition here extend the DefinitelyTyped `sharedb` class, which is the name in those definitions for `sharedb.Backend` in the sharedb JS code.